account/account_test.go

```diff
@@ -344,3 +355,132 @@ func TestConvertGDPREnforcePurposeFields(t *testing.T) {
+			givenConfig:         []byte(`{"gdpr":{"purpose1":{"enforce_purpose":"full"}}}`),
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Old GDPR Purpose 2 Config Is Used, Expect Purpose Metric Method to Be Called",
+			givenConfig:         []byte(`{"gdpr":{"purpose2":{"enforce_purpose":"full"}}}`),
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Old GDPR Purpose 3 Config Is Used, Expect Purpose Metric Method to Be Called",
+			givenConfig:         []byte(`{"gdpr":{"purpose3":{"enforce_purpose":"full"}}}`),
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Old GDPR Purpose 4 Config Is Used, Expect Purpose Metric Method to Be Called",
+			givenConfig:         []byte(`{"gdpr":{"purpose4":{"enforce_purpose":"full"}}}`),
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Old GDPR Purpose 5 Config Is Used, Expect Purpose Metric Method to Be Called",
+			givenConfig:         []byte(`{"gdpr":{"purpose5":{"enforce_purpose":"full"}}}`),
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Old GDPR Purpose 6 Config Is Used, Expect Purpose Metric Method to Be Called",
+			givenConfig:         []byte(`{"gdpr":{"purpose6":{"enforce_purpose":"full"}}}`),
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Old GDPR Purpose 7 Config Is Used, Expect Purpose Metric Method to Be Called",
+			givenConfig:         []byte(`{"gdpr":{"purpose7":{"enforce_purpose":"full"}}}`),
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Old GDPR Purpose 8 Config Is Used, Expect Purpose Metric Method to Be Called",
+			givenConfig:         []byte(`{"gdpr":{"purpose8":{"enforce_purpose":"full"}}}`),
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Old GDPR Purpose 9 Config Is Used, Expect Purpose Metric Method to Be Called",
+			givenConfig:         []byte(`{"gdpr":{"purpose9":{"enforce_purpose":"full"}}}`),
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Old GDPR Purpose 10 Config Is Used, Expect Purpose Metric Method to Be Called",
+			givenConfig:         []byte(`{"gdpr":{"purpose10":{"enforce_purpose":"full"}}}`),
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "New GDPR Purpose Config Is Used, Don't Expect Purpose Metric Method to Be Called",
+			givenConfig:         []byte(`{"gdpr":{"purpose1":{"enforce_algo":"full"}}}`),
+			expectedMetricCount: 0,
+		},
+	}
+
+	for _, test := range testCases {
+		metrics := &metrics.MetricsEngineMock{}
+		metrics.Mock.On("RecordAccountGDPRPurposeWarning", mock.Anything, mock.Anything).Return()
+		metrics.Mock.On("RecordAccountUpgradeStatus", mock.Anything, mock.Anything).Return()
+
+		_, err := ConvertGDPREnforcePurposeFields(test.givenConfig, metrics, "acct-id")
+		assert.NoError(t, err)
+
+		metrics.AssertNumberOfCalls(t, "RecordAccountGDPRPurposeWarning", test.expectedMetricCount)
+		metrics.AssertNumberOfCalls(t, "RecordAccountUpgradeStatus", test.expectedMetricCount)
```

Comment:
Again, are there no cases where `RecordAccountUpgradeStatus` is called or not called independently of the other given metrics? If not, would it be better to just drop this metrics and replace it with a sum over the other metrics?

Reply:
@hhhjort you're right that as of right now `RecordAccountUpgradeStatus` is called every time some other more detailed metric function is called. The plan though is for `RecordAccountUpgradeStatus` to persist long-term but for all of the other field-specific metrics to be replaced by logging once fetchers 2.0 is complete since we'll then be able to perform smarter logging that won't flood the logs. The `RecordAccountUpgradeStatus` metric would then serve as an indicator that you should check your logs for more detailed info about your account config problems.

One question I have at this moment though is whether `RecordAccountUpgradeStatus` should count the number of times an account is referenced that contains at least one account config issue, or if it should count the number of account config issues detected across all of the account references. This PR is currently doing the latter but I'm thinking the former might make more sense.
Thoughts?

For example, assume account A uses one deprecated field and account B uses two deprecated fields.
In the first case, every time account A is referenced we increment by 1 and every time account B is referenced we increment by 1.
In the second case, every time acount A is referenced we increment by 1 and every time B is referenced we increment by 2.

Reply:
I don't have a strong feeling for this either way. I think the important information is which accounts still need to be updated, which really only cares if the number recorded >=1 for a particular account, not what that number turns out to be. Also there would be uncertainty with a high traffic account being loaded more often than a low traffic one, so even with a high number does that mean it has a lot of fields that need updated, or just that the load was called rather often.

Otherwise this PR is looking good.